Sources/BrowserServicesKit/Autofill/WebsiteAutofillUserScript.swift

```diff
@@ -92,11 +109,31 @@ public class WebsiteAutofillUserScript: AutofillUserScript {
         }
         // Sets the last message host, so we can check when it messages back
         lastOpenHost = hostProvider.hostForMessage(message)
-
+                
+        var containsSensitiveData = false
+        
+        if let jsonData = serializedInputContext.data(using: .utf8) {
+            let inputData = try? JSONDecoder().decode(InputData.self, from: jsonData)
+                        
+            if let inputType = inputData?.inputType,
+               let prefix = inputType.split(separator: ".").first {
+                if SecureInputTypes(rawValue: String(prefix)) != nil {
+                    containsSensitiveData = true
+                    
+                    // Do not require Vault access to generated passwords
+                    if (inputData?.credentials?[0].autogenerated) != nil {
```

Comment:
In case `credentials` is empty, this would crash. How about using `first?` here?

```suggestion
                    if (inputData?.credentials?.first?.autogenerated) != nil {
```

Reply:
Good catch!.  This was deleted with the last changes.